controllers/platformnetwork_controller.go

```diff
@@ -1248,6 +1248,14 @@ func (r *PlatformNetworkReconciler) Reconcile(ctx context.Context, request ctrl.
 		return reconcile.Result{}, err
 	}
 
+	// Restore the PlatformNetwork status
+	if r.checkRestoreInProgress(instance) {
+		if err := r.RestorePlatformNetworkStatus(instance); err != nil {
+			return reconcile.Result{}, err
+		}
+		return reconcile.Result{}, err
```

Comment:
Should we rather return `ctrl.Result{}, nil` to be on a safer side, we really don't want to trigger reconciliation at any point after removing the restore-in-progress annotation unless it's coming from the user.
Either way, I'm expecting a test case to verify this.